frontend/config/apollo.js

```diff
@@ -18,7 +18,17 @@ const authMiddleware = new ApolloLink((operation, forward) => {
 
 const client = new ApolloClient({
   link: concat(authMiddleware, httpLink),
-  cache: new InMemoryCache()
+  cache: new InMemoryCache({
+    dataIdFromObject: o => {
+      switch (o.__typename) {
+        case 'User':
+          // The primary key of 'User' is 'username'
```

Comment:
Is is worth it to go back to username's pk equaling an Id? We talked about usernames making more sense weeks ago, but it seems to be causing problems.

Reply:
If it's not too much effort to switch to id's, we could, as Apollo automatically supports objects w/ an `id` or `_id` field; however, I don't think it's any more difficult on the frontend w/ usernames. This config here allows Apollo's store to keep track of how different User update from mutations.